checker/tests/interning/Raw3.java

```diff
@@ -54,7 +54,7 @@
   // with raw types
   List<@Interned String> foo6() {
     List<String> sl = new ArrayList<>();
-    //:: warning: [unchecked] unchecked conversion
+    //:: warning: [unchecked] unchecked conversion :: error: (return.type.incompatible)
```

Comment:
You are saying that we should prefer propagating the type from the cast expression instead of preferring the assignment context, right?
Was the previous behavior that the raw type in the cast used the assignment context? Then because a cast from `List<@UnknownInderned String>` (the type of `s1`) to `List<@Interned String>` (the return type) is legal, the previous behavior allowed this?